src/bare_metal_billing/models.py

```diff
@@ -0,0 +1,57 @@
+from dataclasses import dataclass, field
+from typing import Annotated
+from typing_extensions import Self
+import datetime
+
+import pydantic
+
+
+def parse_date(v: str | datetime.date) -> datetime.date:
+    if isinstance(v, str):
+        return datetime.datetime.fromisoformat(v)
+    return v
+
+
+def parse_bm_accelerator(v: list) -> int:
+    return len(v)
+
+
+PositiveInt = Annotated[int, pydantic.Field(gt=0)]
+DateField = Annotated[datetime.datetime, pydantic.BeforeValidator(parse_date)]
+
+
+class BMResourceProperties(pydantic.BaseModel):
+    cpu_count: Annotated[int, pydantic.Field(alias="cpus", gt=-1)] = 0
+    ram: Annotated[int, pydantic.Field(alias="memory_mb", gt=-1)] = 0
+    gpu_count: Annotated[
+        int,
+        pydantic.Field(alias="accelerators", gt=-1),
+        pydantic.BeforeValidator(lambda v: len(v)),
+    ] = 0
+
+
+class BMNodeUsage(pydantic.BaseModel):
+    resource_class: Annotated[str, pydantic.Field(alias="Resource Class")]
+    resource_properties: Annotated[
+        BMResourceProperties, pydantic.Field(alias="Resource Properties")
+    ]
+    project: Annotated[str, pydantic.Field(alias="Project")]
+    start_time: Annotated[DateField, pydantic.Field(alias="Start Time")]
+    expire_time: Annotated[DateField | None, pydantic.Field(alias="Expire Time")] = None
+
+    @pydantic.model_validator(mode="after")
+    @classmethod
+    def validate_expire_time(cls, data: Self):
+        if data.expire_time and data.expire_time < data.start_time:
+            raise ValueError("expire_time must be after start_time")
+        return data
+
+
+class BMUsageData(pydantic.RootModel):
+    root: list[BMNodeUsage]
+
+
+@dataclass
```

Comment:
Why are we mixing in a `dataclass` in a module in which we're already using `pydantic` models?

Reply:
I have turned `ProjectUsage` into a `pydantic` module